src/dali/plugin/input/csv/ledger.py

```diff
@@ -0,0 +1,134 @@
+from dali.intra_transaction import IntraTransaction
+
+_SENT: str = "OUT"
+_RECV: str = "IN"
+
+
+class InputPlugin(AbstractInputPlugin):
+
+    __LEDGER: str = "Ledger"
+
+    __TIMESTAMP_INDEX: int = 0
+    __CURRENCY_INDEX: int = 1
+    __OPERATION_TYPE_INDEX: int = 2
+    __QUANTITY_INDEX: int = 3
+    __FEE_INDEX: int = 4
+    __TRANSACTION_ID_INDEX: int = 5
+
+    __DELIMITER = ","
+
+    __CURRENCY_ALIAS_DICT = {
+        "FANTOM": "FTM",
+    }
+
+    def __init__(
+        self,
+        account_holder: str,
+        account_nickname: str,
+        csv_file: str,
+        native_fiat: Optional[str] = None,
+    ) -> None:
+
+        super().__init__(account_holder=account_holder, native_fiat=native_fiat)
+        self.__account_nickname: str = account_nickname
+        self.__csv_file: str = csv_file
+        self.__logger: logging.Logger = create_logger(f"{self.__LEDGER}/{self.__account_nickname}/{self.account_holder}")
+
+    def load(self) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(self.__csv_file, encoding="utf-8") as csv_file:
+            lines = reader(csv_file, delimiter=self.__DELIMITER)
+            header_found: bool = False
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                if not header_found:
+                    # Skip header line
+                    header_found = True
+                    self.__logger.debug("Header: %s", raw_data)
+                    continue
+                timestamp: str = line[self.__TIMESTAMP_INDEX]
+                timestamp_value: datetime = dateutil.parser.isoparse(timestamp)  # For example, "2022-06-05T00:39:14.007Z"
+                self.__logger.debug("Transaction: %s", raw_data)
+                currency: str = line[self.__CURRENCY_INDEX]
+                currency = self.__CURRENCY_ALIAS_DICT.get(currency, currency)
+                transaction_type: str = line[self.__OPERATION_TYPE_INDEX]
+                spot_price: str = Keyword.UNKNOWN.value
+                crypto_hash: str = line[self.__TRANSACTION_ID_INDEX]
+                fee_str: str = line[self.__FEE_INDEX]
+                fee_number: RP2Decimal = RP2Decimal(fee_str) if fee_str else ZERO  # Fee is sometimes missing
+                quantity_number: RP2Decimal = RP2Decimal(line[self.__QUANTITY_INDEX])
+
+                if quantity_number == ZERO and fee_number > ZERO:
+                    self.__logger.warning("Possible dusting attack (fee > 0, total amount = 0): %s", raw_data)
+                    continue
+                if transaction_type in {_SENT, _RECV}:  # Need example data for sent transactions, untested as of 7/9/2022
+                    result.append(
+                        IntraTransaction(
+                            plugin=self.__LEDGER,
+                            unique_id=crypto_hash,
+                            raw_data=raw_data,
+                            timestamp=f"{timestamp_value}",
```

Comment:
Is this generated timestamp in [ISO8601 format](https://github.com/eprbell/dali-rp2/blob/main/docs/user_faq.md#what-is-the-timestamp-format)? Based on what you documented in line 16 I think so, but just wanted to double-check.

Reply:
The timestamp in the Ledger CSV file appears to be ISO8601 format.  The date and time are always separated by "T", there is always a 3 digit millisecond fraction, and always a "Z" time zone designator, for example:
2022-06-05T00:39:14.007Z

The timezone string is parsed by the dateutil module on line 99, producing a datetime. Here we explicitly expect ISO8601 format.

The datetime is converted back to a string on line 119, but this is not ISO8601 format.  It will look like this:
2022-06-05 00:39:14.007000+00:00

Then, the AbstractTransaction uses the dateutil module to parse the string again without assuming any particular format.